includes/ImagePage.php

```diff
@@ -515,22 +515,33 @@ protected function openShowImage() {
 			}
 
 			if ( $showLink ) {
-				$filename = wfEscapeWikiText( $this->displayImg->getName() );
-				$linktext = $filename;
+				$linktext = htmlspecialchars( $this->displayImg->getName(), ENT_QUOTES );
+				$imgTitle = $this->displayImg->getTitle();
 				if ( isset( $msgbig ) ) {
-					$linktext = wfEscapeWikiText( $msgbig );
+					$linktext = htmlspecialchars( $msgbig, ENT_QUOTES );
 				}
-				$medialink = "[[Media:$filename|$linktext]]";
+				$medialink = Linker::makeMediaLinkFile( $imgTitle, $this->displayImg, $linktext );
 
 				if ( !$this->displayImg->isSafeFile() ) {
-					$warning = wfMsgNoTrans( 'mediawarning' );
-					$wgOut->addWikiText( <<<EOT
+					$warning = wfMessage( 'mediawarning' )->parse();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$warning" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)